unstructured/partition/auto.py

```diff
@@ -199,6 +199,8 @@ def partition(
             detect_language_per_element
                 Detect language per element instead of at the document level.
     pdf_infer_table_structure
+        Deprecated! Use `skip_infer_table_types` to opt out of table extraction for any document
+        type.
```

Comment:
lets add a warning log about this deprecated setting

Reply:
I've made it so that the warning is raised when we know someone explicitly used the parameter (so when it's `False`). I am not sure quite if that's effective, e.g. if users mostly do `pdf_infer_table_structure=True` they won't realize it's better to let go of that practice. Also wouldn't want to spam warning every time.

What do you think? Technically we could set it `None` on default, so that would detect every time someone provides value explicitly but maybe made things more confusing, loose thought.